packages/ai/src/ai/common/chat.py

```
@@ -451,6 +478,86 @@ def chat(self, question: Question) -> Answer:
+        """
+        from ai.constants import CONST_CHAT_MAX_RETRIES, CONST_CHAT_BASE_DELAY, CONST_CHAT_MAX_DELAY
+
+        max_network_retries = CONST_CHAT_MAX_RETRIES
+        base_delay = CONST_CHAT_BASE_DELAY
+        max_delay = CONST_CHAT_MAX_DELAY
+
+        for attempt in range(max_network_retries):
+            emitted = {'value': False}
+
+            def gated(text: str) -> None:
+                emitted['value'] = True
```

Ruff (0.15.12)

[warning] Function definition does not bind loop variable `emitted`

(B023)